client/post-editor/media-modal/image-editor/image-editor-canvas.jsx

```diff
@@ -63,17 +64,17 @@ const MediaModalImageEditorCanvas = React.createClass( {
 	},
 
 	componentWillReceiveProps( newProps ) {
-		if ( this.props.src !== newProps.src ) {
+		if ( this.state.src !== newProps.src ) {
 			this.getImage( newProps.src );
 		}
 	},
 
 	componentDidMount() {
-		if ( ! this.props.src ) {
+		if ( ! this.state.src || ( this.state.src === this.props.src ) ) {
```

Comment:
This is probably the logic error for the infinite placeholder. We should let the image load once on mount, even if it's the same img src.